java.md

```diff
@@ -178,11 +178,62 @@ to manage each platform specific Jar.  Release individual architecture specific
 instance to download these released jars and package them into a combined Jar with a final `mvn release:perform`.
 An example of this methd can be found in the [Leveldbjni-native](https://github.com/fusesource/leveldbjni) `pom.xml` files. 
 
+### Remove Anti-patterns
+
+Anti-patterns can affect the performance on any instance family, but the level
+of impact can be different.  Below is a list of
+anti-patterns we have found to be particularly impactful on Graviton:
+
+1.  **Excessive exceptions**: Throwing exceptions and generating stack-traces
+has been observed to cost up to 2x more on Graviton platforms compared to x86.
+We recommend not to use Java exceptions as control flow, and to remove
+exceptions when they appear in the hot-code path. Identifying hot exceptions can
+be done using function profilers like [Aperf](https://github.com/aws/aperf), 
+[Async-profiler](https://github.com/async-profiler/async-profiler), or Linux `perf`.
+Overhead can be mitigated some by using the `-XX:+OmitStackTraceInFastThrow` JVM
+flag to allow the Java runtime to optimize the exception flow for some hot
+paths. The best solution is to avoid the exceptions as much as possible.
 
 ### Profiling Java applications
-For languages that rely on a JIT (such an Java), the symbol information that is
-captured is lacking, making it difficult to understand where runtime is being consumed.
-Similar to the code profiling example above, `libperf-jvmti.so` can be used to dump symbols for
+
+### APerf + Async-profiler
+
+To profile Java we recommend using [Aperf](https://github.com/aws/aperf) to gather profiles and view them
+via a static webpage.  On your test system, follow the below directions to profile your
+Java code using APerf:
+
+```bash
+# Get latest APerf release onto the machine you are profiling.  
+# As of Oct 28, 2024 the latest release is v0.13.0.
+wget https://github.com/aws/aperf/releases/download/v0.1.13-alpha/aperf-v0.1.13-alpha-aarch64.tar.gz
+tar -zxf aperf-v0.1.13-alpha-aarch64.tar.gz
+
+# Get the latest Async profiler
+wget https://github.com/async-profiler/async-profiler/releases/download/v3.0/async-profiler-3.0-linux-arm64.tar.gz
+tar -zxf async-profiler-3.0-linux-arm64.tar.gz
+cd async-profiler-3.0-linux-arm64
+sudo mkdir -p /opt/bin
+sudo mkdir -p /opt/lib
+sudo cp -a bin/* /opt/bin
+sudo cp -a lib/* /opt/lib
```

Comment:
It seems unnecesaary to get putting these in system paths if you're still going to require setting the PATH to run aperf. I'd say leave them where they are and set the PATH to the extracted location. It doesn't clutter up random file system paths that way.

It's odd that async-profiler doesn't have a recommended usage, particularly given that it has library dependencies (so requires a correct library path, too).